app/models/asset.rb

```diff
@@ -76,7 +76,7 @@ class Asset
 
   mount_uploader :file, AssetUploader
 
-  before_save :store_metadata, unless: :uploaded?
+  before_save :store_metadata, unless: :uploaded?, if: -> { changes.include?(:file) }
```

Comment:
I think this might help reduce the number of raised errors. 

It wouldn't if the second process is already inside the store_metadata method